nanodbc/nanodbc.cpp

```diff
@@ -2062,12 +2055,31 @@ void statement::statement_impl::bind_strings(
         }
     }
 
-    auto const buffer_length = value_size * sizeof(string::value_type);
-    bound_buffer<string::value_type> buffer(values, batch_size, buffer_length);
+    auto const buffer_length = value_size * sizeof(T);
+    bound_buffer<T> buffer(values, batch_size, buffer_length);
     bind_parameter(param, buffer);
 }
 
 template <>
+bool statement::statement_impl::equals(const std::string& lhs, const std::string& rhs)
+{
+    return std::strncmp(lhs.c_str(), rhs.c_str(), lhs.size()) == 0;
+}
+
+template <>
+bool statement::statement_impl::equals(const wide_string& lhs, const wide_string& rhs)
+{
+    // convert narrow strings, cause wcsncmp doesn't work on some machines
```

Comment:
machines? You mean compilation toolkits? Any example?

Reply:
I've looked at the git log why not simply wcsncmp was used and the related commit message was:
e6059ff3a79062f83256b9d1d3c9c8368798781e
> Functions like `swprintf()`, `wcsftime()`, `wcsncmp()` can not be used
> with `u16string` types. Instead, prefers to narrow unicode string to
> work with them, and then widen them after work has been completed.

I will change the code comment to this so future readers will understand it better.

Reply:
Thanks for looking at this. Good idea about the comment.